specs/rollup-node.md

```diff
@@ -251,6 +253,46 @@ However, they can still be challenged by a fault proof until the end of the faul
 
 > **TODO** the spec doesn't encode the notion of fault proof yet, revisit this (and include links) when it does
 
+## From L2 Block to L2 Output Root
+
+After processing a block the resulting outputs will need to be synchronized with L1 for trustless execution of
+L2-to-L1 messaging, such as withdrawals. To synchronize outputs are merkleized: hashed in a structured form for minimal
+proof cost to any piece of data. The merkle-structure is defined with [SSZ], a type system for merkleization and
+serialization, used in L1 (beacon-chain). However, we replace `sha256` with `keccak256` to save gas costs in the EVM.
```

Comment:
How easy is it to maintain our own version of SSZ?

Reply:
To swap the hash-function, you need to change two things in most implementations:
- the `hasher` object you pass as argument (it's an argument, and not a function call, since it is re-used a lot during tree hashing, like a scratch-pad)
- the `zero_hashes` (`H_1 = H(0, 0), H_2 = H(H_1, H_1), ...`) need to be initialized with the desired hash-function

These are pretty simple changes, and they seem worth the gas-cost savings to me.